lib/vagrant-service-manager/plugin.rb

```diff
@@ -8,6 +11,15 @@ class Plugin < Vagrant.plugin('2')
         require_relative 'command'
         Command
       end
+
+      config 'servicemanager' do
+        require_relative 'config'
+        Config
+      end
+
+      action_hook(:servicemanager, :machine_action_up) do |hook|
```

Comment:
After adding the guest detection capability, this is resolved. Now, it will only run against supported [ADB / CDK] boxes and `docker` is the default provider.